### ISSUE TITLE
StackLLAMA script to merge peft adapter does not work with checkpoints

### ISSUE THREAD

**Issue by jvhoffbauer** [lvwerra/trl#609]
When using the Huggingface Trainer with a model that has LoRA adapters, the checkpoints stored contain only the adapters. The complete model is not store, it is only stored when I call 
```model.save_pretrained(output_name + "_peft_last_checkpoint")```

It should be possible to merge a model from arbitrary checkpoints. 

Happy to provide a PR if that helps.

**Comment by jvhoffbauer** [comment]
I understand. I was unclear in terms of what the issue is. I'll try to elaborate. 

When I want to use a checkpoint that the `SFTTrainer` generated, the merging script does not work. It generates a directory containing the base model without any merged adapters.  

```
python3 merge_peft_adapter.py \
    --adapter_model_name=checkpoint-2500 \
    --base_model_name=meta-llama/Llama-2-7b-chat-hf 
```

This is unexpected because I would either expect it to merge the model based on the checkpoint or raise an error.

**Comment by lvwerra** [comment]
Not sure what the issue is here. How did you verify that the folder contains the base model and not the new model with the merged adapters? In general you could also add a `model = model.merge_and_unload()` before you save the model in the SFTTrainer script.

**Comment by jvhoffbauer** [comment]
You're right, this is not an issue. The new script should work both with checkpoints saved by the trainer as well as with checkpoints saved manually. 

I confused the new and old script!
